portfolio/src/main/webapp/script.js

```diff
@@ -13,16 +13,63 @@
 // limitations under the License.
 
 /**
- * Adds a random greeting to the page.
+ * Adds a random fact to the page.
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function addRandomFact() {
+  const facts =
+      ['🏙️: I am from Mexico City (capital of Mexico)', '🏎️: I am an avid Formula 1 fan',
+       '🏫: I graduated from highschool a year early', 
+       '🎓: I was at Stanford for a year with both of my older siblings',
+       '🚣‍♂️: I was a competitive rower in high school'];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+  // Pick a random fact.
+  const fact = facts[Math.floor(Math.random() * facts.length)];
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+  // Add the fact tp the stage.
+  const factContainer = document.getElementById('fact-container');
+  factContainer.innerText = fact;
 }
+
+let i = 0;
+let j = 0;
+
+const PHRASES = ["software engineer 🖥️", "designer 🎨", "product manager 💼", 
+               "photographer 📷", "proud Mexican 🇲🇽", "entrepreneur 📈"];
+const RATE = 100;
+const PAUSE = 1000;
+
+/**
+ * Fills in the 'I am a' sentences with different phrases
+ */
+function writeSnippets() {
+    if (i == 0) document.getElementById("text").innerHTML = "I am a ";
+	if (i < PHRASES[j].length) {
+        console.log(i)
+        document.getElementById("text").innerHTML += PHRASES[j].charAt(i);
+		i++;
+		setTimeout(writeSnippets, RATE);
```

Comment:
did you check out requestAnimationFrame?
https://medium.com/javascript-in-plain-english/better-understanding-of-timers-in-javascript-settimeout-vs-requestanimationframe-bf7f99b9ff9b seems relevant

Reply:
I'll check it out and try messing around with requestAnimationFrame